src/Driver/ParallelFilesystemDriver.php

```diff
@@ -29,11 +29,16 @@ final class ParallelFilesystemDriver implements FilesystemDriver
     private Future $pendingWorker;
 
     /**
-     * @param int       $workerLimit Maximum number of workers to use from the pool for open files.
+     * @param WorkerPool|null $pool Custom worker pool to use for file workers. If null, a new pool is created.
+     * @param int $workerLimit Maximum number of workers to use from the pool for open files.
      */
     public function __construct(?WorkerPool $pool = null, int $workerLimit = self::DEFAULT_WORKER_LIMIT)
     {
-        $this->pool = $pool ?? workerPool();
+        if ($workerLimit <= 0) {
+            throw new \ValueError("Worker limit must be a positive integer");
+        }
+
+        $this->pool = $pool ?? new ContextWorkerPool($workerLimit);
```

Comment:
Shouldn't we deprecate the `$workerLimit` parameter? We could remove the worker limit from the implementation then, because we have a dedicated pool usually. Actually... shouldn't using a subset of an existing pool be its own pool implementation if really needed?

Reply:
Should we emit a deprecation notice if the parameter is passed?

Reply:
I think so, yes. I'm looking at adding the pool implementation you suggested above to parallel as well.